packages/curve-ui-kit/src/shared/ui/PositionDetails/HealthBar.tsx

```diff
@@ -0,0 +1,102 @@
+import { Box, Typography, type Theme } from '@mui/material'
+import { t } from '@ui-kit/lib/i18n'
+import { Reds, Blues } from '@ui-kit/themes/design/0_primitives'
+
+type HealthBarProps = {
+  health: number | undefined | null
+}
+
+const BAR_HEIGHT = '1.4375rem' // 23px
+const LINE_WIDTH = '0.25rem' // 4px
+const LABEL_GAP = '0.125rem' // 2px
+
+type LineColor = 'red' | 'orange' | 'green' | 'dark-green'
+
+const getLineColor = (color: LineColor) => (t: Theme) => {
+  switch (color) {
+    case 'red':
+      return t.design.Color.Tertiary[600]
+    case 'orange':
+      return t.design.Color.Tertiary[400]
+    case 'dark-green':
+      return t.design.Color.Secondary[600]
+    case 'green':
+      return t.design.Color.Secondary[500]
+  }
+}
```

Comment:
Maybe a matter of taste but I think this is more readable
```suggestion
const getLineColor = (color: LineColor) => (t: Theme) => ({
  red: t.design.Color.Tertiary[600],
  orange: t.design.Color.Tertiary[400],
  ['dark-green']: t.design.Color.Secondary[600],
  green: t.design.Color.Secondary[500]
}[color])
```